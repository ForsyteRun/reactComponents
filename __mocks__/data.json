{
  "status": 200,
  "kind": "books#volumes",
  "totalItems": 1392,
  "items": [
    {
      "id": "pfaeBAAAQBAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "8cm3DwAAQBAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "xcRiAAAAQBAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "NmseEAAAQBAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "8ZeGEAAAQBAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "c-g2DwAAQBAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "0HkVEAAAQBAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "5_cwDwAAQBAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "xJOnEAAAQBAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "qer7ic8z03IC",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "aW6fEAAAQBAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "Y0yFEAAAQBAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "ma8VEAAAQBAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "6hzsWwz1aLAC",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "4ec6AQAAMAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "37ieAAAAMAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "v-ynEAAAQBAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "kTwpAAAAYAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "vjkEAQAAIAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "Q1BpAAAAIAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "cTZW0qg-uecC",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "FvXZAAAAMAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "U-kaAAAAYAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "hd0mAQAAMAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "SxcJAAAAIAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },

    {
      "id": "M0sbAAAAYAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "f15qAAAAMAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "xT7xAAAAMAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "DnBPAAAAYAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "vXRHAAAAIAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "gTUm9j-jLVAC",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "SJ9PAQAAIAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "C9RKAQAAMAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "s4FtAAAAMAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "cbxNAQAAIAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "E-hPAAAAYAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "NasbAQAAMAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "EtIdAAAAMAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "o9tqAAAAMAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    },
    {
      "id": "3zWQAAAAMAAJ",
      "volumeInfo": {
        "title": "Disorders of Sex Development in Gynaecology (Russian edition)",
        "authors": ["Zograb Makiyan"],
        "pageCount": 167,
        "imageLinks": {
          "thumbnail": "http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
        },
        "language": "ru"
      }
    }
  ]
}
