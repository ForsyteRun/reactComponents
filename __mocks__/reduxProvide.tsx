import { ThunkMiddleware, configureStore } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import { booksApi } from '../src/services/fetchData';
import paginationSlice from './../src/store/slices/pagination';
import searchSlice from './../src/store/slices/search';
import cardSlice from './../src/store/slices/card';
import loadingSlice from './../src/store/slices/loading';

export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    store = configureStore({
      reducer: {
        [booksApi.reducerPath]: booksApi.reducer,
        search: searchSlice,
        pagination: paginationSlice,
        card: cardSlice,
        loading: loadingSlice,
      },
      middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware().concat(booksApi.middleware as ThunkMiddleware),
      preloadedState,
    }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <BrowserRouter>
        <Provider store={store}>{children}</Provider>;
      </BrowserRouter>
    );
  }

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
