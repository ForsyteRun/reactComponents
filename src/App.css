#root {
  max-width: 330px;
  margin: 0 auto;
  text-align: center;
}

a {
  text-decoration: none;
  cursor: pointer;
}

button {
  cursor: pointer;
}

.title {
  margin-bottom: 1rem;
}

form {
  display: flex;
  flex-direction: column;
  align-items: center;
  background: #f5f0f0;
  border-radius: 0.51rem;
  box-shadow: rgba(0, 0, 0, 0.24) 0px 3px 8px;
  padding: 1rem;
}

form > label {
  margin-bottom: 1rem;
}

form > label > input {
  border: 0.5px solid transparent;
  border-radius: 2px;
  padding: 0.2rem;
}

form > button {
  width: 100%;
  border-radius: 0.2rem;
  background: #6d2bd6;
  color: aliceblue;
  border: transparent;
  padding: 0.5rem 0.3rem;
  box-shadow: rgba(0, 0, 0, 0.773) 0.3rem 3px 0.4rem;
  transition: all 0.1s ease-out;
  cursor: pointer;
}

form > button:active {
  transform: translate(2px, 2px);
  transition: all 0.1s ease-out;
}

form > button:disabled {
  background: #4a494d;
  color: rgb(205, 209, 213);
  transform: none;
}

.upload {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-end;
  border-bottom: 1px solid #9999997c;
  padding: 0.5rem 0;
}

.upload > h5 {
  align-self: normal;
}

h5 {
  margin: 0;
  padding: 0;
  color: red;
  font-size: 0.6rem;
}

.inputContainer {
  height: 3rem;
}

.easyLevel {
  color: rgb(238, 58, 8);
}

.normalLevel {
  color: rgb(34, 0, 128);
}

.hardLevel {
  color: rgb(0, 128, 0);
}

.containerPasswordLevel {
  position: absolute;
  right: -3.5rem;
  top: 0;
  font-weight: 700;
}
