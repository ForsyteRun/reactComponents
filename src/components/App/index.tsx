import React from 'react';
import { StateType } from '../../types';
import s from './App.module.css';
import ListItems from '../ListItem';
import Header from '../Header';
class App extends React.Component<NonNullable<unknown>, StateType> {
  state = {
    count: 2110,
    next: 'https://pokeapi.co/api/v2/item?offset=10&limit=10',
    previous: null,
    results: null,
  };

  componentDidMount = async () => {
    try {
      // const storageData = localStorage.getItem('data');
      const response: Response = await fetch('https://pokeapi.co/api/v2/item');
      const data: StateType = await response.json();
      this.setState(data);
    } catch (error) {
      throw new Error(`error: ${error}`);
    }
  };

  render() {
    return (
      <div className={s.container}>
        <Header />
        <ListItems results={this.state.results} />
      </div>
    );
  }
}

export default App;
