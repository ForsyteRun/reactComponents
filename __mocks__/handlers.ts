import { http, HttpResponse } from 'msw';
import { IFetchData } from '../src/types';
import data from './data.json';

const jsonData: IFetchData = data;

export const handlers = [
  http.get('https://www.googleapis.com/books/v1/volumes', () => {
    return HttpResponse.json(jsonData.items);
  }),

  http.get('https://www.googleapis.com/books/v1/volumes/:id', (req) => {
    const { id } = req.params;

    const foundItem = jsonData.items.find((item) => item.id === id);

    return HttpResponse.json(foundItem);
  }),
];
