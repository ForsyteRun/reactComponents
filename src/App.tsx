import React from 'react';
import './App.css';
import { StateType } from './types';

class App extends React.Component<NonNullable<unknown>, StateType> {
  constructor(props: NonNullable<unknown>) {
    super(props);
    this.state = {
      count: 2110,
      next: 'https://pokeapi.co/api/v2/item?offset=10&limit=10',
      previous: null,
      results: null,
    };
  }

  getData = async () => {
    try {
      const response: Response = await fetch('https://pokeapi.co/api/v2/item');
      const data: StateType = await response.json();
      this.setState(data);
    } catch (error) {
      throw new Error(`error: ${error}`);
    }
  };

  render() {
    return (
      <div>
        Hello
        <button onClick={this.getData}>Click me!</button>
        {console.log(this.state.results) as React.ReactNode}
      </div>
    );
  }
}

export default App;
