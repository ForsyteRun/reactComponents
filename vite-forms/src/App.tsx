import './App.css';

function App() {
  let a: any;
  return <div>INIt</div>;
}

export default App;
