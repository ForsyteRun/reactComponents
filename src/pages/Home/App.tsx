import { useCallback, useEffect, useState } from 'react';
import { useLoaderData } from 'react-router-dom';
import s from './App.module.css';
import { IFetchData, IItem } from '../../types';
import { fetchData } from '../../loaders';
import { ListItems, Pagination, Search } from '../../components';
import Select from '../../components/Select';

const App = () => {
  const [loading, setLoading] = useState<boolean>(false);
  const [query, setQuery] = useState<string>('');
  const [books, setBooks] = useState<IItem[]>([]);
  const [error, setError] = useState<boolean>(false);
  const [pageNumber, setPageNumber] = useState<number>(1);
  const [itemsPerPage, setItemsPerPage] = useState<number>(10);

  const fetchInitData = useLoaderData() as IFetchData;

  if (error) {
    throw new Error('Error');
  }

  useEffect(() => {
    async function getData() {
      try {
        setLoading(true);
        const data = await fetchData(query, pageNumber, itemsPerPage);

        if (!data?.items) {
          setLoading(false);
          return;
        }

        setBooks(data.items);
        setLoading(false);
      } catch (error) {
        setLoading(false);
        throw new Error('error' + error);
      }
    }

    getData();
  }, [query, pageNumber, itemsPerPage]);

  useEffect(() => {
    setBooks(fetchInitData.items);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const handleError = useCallback(() => {
    setError(true);
  }, []);

  return (
    <div className={s.container}>
      <Search setQuery={setQuery} />
      <button onClick={handleError}>get error</button>
      <Select setItemsPerPage={setItemsPerPage} />
      {loading ? (
        <div className="lds-dual-ring"></div>
      ) : books ? (
        <>
          <ListItems items={books} />
          <Pagination pageNumber={pageNumber} setPageNumber={setPageNumber} />
        </>
      ) : (
        <div>Not found book with name {query ? query : 'Noname'}</div>
      )}
    </div>
  );
};

export default App;
