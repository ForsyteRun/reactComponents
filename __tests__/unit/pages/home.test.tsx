/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom';
import React from 'react';
import { renderWithProviders } from '../../../__mocks__/reduxProvide';
import App from '../../../pages/index';

global.React = React;
jest.mock('next/router', () => require('next-router-mock'));
jest.mock('next/navigation', () => {
  return {
    useSearchParams: jest.fn(() => ({
      get: jest.fn(),
    })),
  };
});

describe('Home page', () => {
  it('should match snapshot', () => {
    const value = {
      books: {
        data: {
          items: [
            {
              id: 'pfaeBAAAQBAJ',
              volumeInfo: {
                title:
                  'Disorders of Sex Development in Gynaecology (Russian edition)',
                authors: ['Zograb Makiyan'],
                pageCount: 167,
                imageLinks: {
                  thumbnail:
                    'http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api',
                },
                language: 'ru',
              },
            },
            {
              id: '8cm3DwAAQBAJ',
              volumeInfo: {
                title:
                  'Disorders of Sex Development in Gynaecology (Russian edition)',
                authors: ['Zograb Makiyan'],
                pageCount: 167,
                imageLinks: {
                  thumbnail:
                    'http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api',
                },
                language: 'ru',
              },
            },
          ],
          kind: 'test',
          totalItems: 100,
        },
        singleBook: {
          id: 'pfaeBAAAQBAJ',
          volumeInfo: {
            title:
              'Disorders of Sex Development in Gynaecology (Russian edition)',
            authors: ['Zograb Makiyan'],
            pageCount: 167,
            imageLinks: {
              thumbnail:
                'http://books.google.com/books/content?id=pfaeBAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api',
            },
            language: 'ru',
          },
        },
      },
      loading: {
        isLoading: false,
      },
    };
    const { asFragment } = renderWithProviders(<App value={value} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
