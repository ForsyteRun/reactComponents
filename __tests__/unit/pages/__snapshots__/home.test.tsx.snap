// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <form
      role="search"
    >
      <input
        name="search"
        type="text"
        value="nature"
      />
      <button
        type="submit"
      >
        search
      </button>
    </form>
    <button>
      get error
    </button>
    <select>
      <option
        value="5"
      >
        5
      </option>
      <option
        value="10"
      >
        10
      </option>
      <option
        value="15"
      >
        15
      </option>
      <option
        value="20"
      >
        20
      </option>
      <option
        value="25"
      >
        25
      </option>
    </select>
    <div
      style="display: flex; flex-direction: row;"
    >
      <div
        class="container"
      >
        <div
          class="containerList"
        >
          <ul>
            <li
              class="item"
            >
              <a
                href="/pfaeBAAAQBAJ/details?"
              >
                <img
                  alt="Disorders of Sex Development in Gynaecology (Russian edition)"
                  data-nimg="1"
                  decoding="async"
                  height="130"
                  loading="eager"
                  src="/_next/image?url=http%3A%2F%2Fbooks.google.com%2Fbooks%2Fcontent%3Fid%3DpfaeBAAAQBAJ%26printsec%3Dfrontcover%26img%3D1%26zoom%3D1%26edge%3Dcurl%26source%3Dgbs_api&w=256&q=75"
                  srcset="/_next/image?url=http%3A%2F%2Fbooks.google.com%2Fbooks%2Fcontent%3Fid%3DpfaeBAAAQBAJ%26printsec%3Dfrontcover%26img%3D1%26zoom%3D1%26edge%3Dcurl%26source%3Dgbs_api&w=128&q=75 1x, /_next/image?url=http%3A%2F%2Fbooks.google.com%2Fbooks%2Fcontent%3Fid%3DpfaeBAAAQBAJ%26printsec%3Dfrontcover%26img%3D1%26zoom%3D1%26edge%3Dcurl%26source%3Dgbs_api&w=256&q=75 2x"
                  style="color: transparent;"
                  width="100"
                />
              </a>
              <div
                class="content"
              >
                <h3>
                  <span
                    class="title"
                  >
                    name:
                  </span>
                   Disorders of Sex Development in Gynaecology (Russian edition)
                  <span>
                    Disorders of Sex Development in Gynaecology (Russian edition)
                  </span>
                </h3>
                <h3>
                  <span
                    class="title"
                  >
                    authors:
                  </span>
                   Zograb Makiyan
                  <span>
                    Zograb Makiyan
                  </span>
                </h3>
                <h3>
                  <span
                    class="title"
                  >
                    language:
                  </span>
                  <span>
                    ru
                  </span>
                </h3>
                <h3>
                  <span
                    class="title"
                  >
                    pageCount:
                  </span>
                  <span>
                    167
                  </span>
                </h3>
              </div>
            </li>
            <li
              class="item"
            >
              <a
                href="/8cm3DwAAQBAJ/details?"
              >
                <img
                  alt="Disorders of Sex Development in Gynaecology (Russian edition)"
                  data-nimg="1"
                  decoding="async"
                  height="130"
                  loading="eager"
                  src="/_next/image?url=http%3A%2F%2Fbooks.google.com%2Fbooks%2Fcontent%3Fid%3DpfaeBAAAQBAJ%26printsec%3Dfrontcover%26img%3D1%26zoom%3D1%26edge%3Dcurl%26source%3Dgbs_api&w=256&q=75"
                  srcset="/_next/image?url=http%3A%2F%2Fbooks.google.com%2Fbooks%2Fcontent%3Fid%3DpfaeBAAAQBAJ%26printsec%3Dfrontcover%26img%3D1%26zoom%3D1%26edge%3Dcurl%26source%3Dgbs_api&w=128&q=75 1x, /_next/image?url=http%3A%2F%2Fbooks.google.com%2Fbooks%2Fcontent%3Fid%3DpfaeBAAAQBAJ%26printsec%3Dfrontcover%26img%3D1%26zoom%3D1%26edge%3Dcurl%26source%3Dgbs_api&w=256&q=75 2x"
                  style="color: transparent;"
                  width="100"
                />
              </a>
              <div
                class="content"
              >
                <h3>
                  <span
                    class="title"
                  >
                    name:
                  </span>
                   Disorders of Sex Development in Gynaecology (Russian edition)
                  <span>
                    Disorders of Sex Development in Gynaecology (Russian edition)
                  </span>
                </h3>
                <h3>
                  <span
                    class="title"
                  >
                    authors:
                  </span>
                   Zograb Makiyan
                  <span>
                    Zograb Makiyan
                  </span>
                </h3>
                <h3>
                  <span
                    class="title"
                  >
                    language:
                  </span>
                  <span>
                    ru
                  </span>
                </h3>
                <h3>
                  <span
                    class="title"
                  >
                    pageCount:
                  </span>
                  <span>
                    167
                  </span>
                </h3>
              </div>
            </li>
          </ul>
          <div
            class="pagination"
          >
            <a
              class="active"
            >
              1
            </a>
            <a>
              2
            </a>
            <a>
              3
            </a>
            <a>
              4
            </a>
            <a>
              5
            </a>
            <a>
              6
            </a>
            <a>
              7
            </a>
            <a>
              8
            </a>
            <a>
              9
            </a>
            <a>
              10
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
